source/UIAHandler/__init__.py

```diff
@@ -1376,3 +1405,14 @@ def terminate():
 
 def _isDebug():
 	return config.conf["debugLog"]["UIA"]
+
+
+WM_NVDA_UIA_FLUSH = winUser.registerWindowMessage("WM_NVDA_UIA_FLUSH")
+
+
+def eventLimiterWindowProc(msg, wParam, lParam):
+	if msg == WM_NVDA_UIA_FLUSH:
+		if lParam == 0:
+			NVDAHelper.localLib.rateLimitedUIAEventHandler_flush(wParam)

```

Comment:
On one hand, doing this in the main thread simplifies things, since threading issues can be tricky to debug. On the other hand, this means we're now doing things like IsUIAWindow on the main thread instead of a background thread. I guess IAccessibleHandler was already doing this, but I wanted to check whether this is something you considered and your reasons for not believing it to be a concern.

Reply:
We could set up a window in a dedicated background thread. But if isUIAWindow did hang, processing of all UIA events would also pause, as they all went via that thread. So I don't think this is much better than it being on the main thread. It is still worth separately considering making isUIAWindow execute on a background thread, but  perhaps something a little bit more involved where we have a thread per UI thread, so that if the UI in one thread freezes, it does not block another one. I think that might be separate to this pr.

Reply:
There is probably lower hanging fruit with isUIAWindow anyway. For example, I wonder whether we should increase the cache time for UIA windows. It's also easier to manage hangs with the watchdog if they're on the main thread.